airbyte-integrations/connectors/source-stripe/source_stripe/error_mappings/stripe_error_mapping.py

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.streams.http.error_handlers.default_error_mapping import DEFAULT_ERROR_MAPPING
+from airbyte_cdk.sources.streams.http.error_handlers.response_models import ErrorResolution, ResponseAction
+
+STRIPE_ERROR_MAPPING = DEFAULT_ERROR_MAPPING | {
+    429: ErrorResolution(
```

Comment:
Why are we overriding this? It seems like [we retry on RATE_LIMIT](https://github.com/airbytehq/airbyte/blob/889cb65e21079ed7aa6b97d540112d4e2ff71c6e/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L317-L329) and [the trace message](https://github.com/airbytehq/airbyte/blob/889cb65e21079ed7aa6b97d540112d4e2ff71c6e/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L281-L292) would probably be a nice addition. 

If we really can't, I think it would be worth documenting why

Reply:
Yes, I was missing adding a back-off strategy to make it work with max_retries. The default strategy returns None, which will make it fall into [retry_endlessly](https://github.com/airbytehq/airbyte/blob/aa8c125e7c9053daec654bcdb491f19ef927eb1b/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L320-L331). I picked an exponential one, as the documentation suggests. Thanks.
https://docs.stripe.com/error-codes#rate-limit
aldogonzalez8/source/stripe/upgrade-cdk3